### ISSUE TITLE
GossipFrame_GetTitleButtonCount when clicking on a pet tamer

### ISSUE THREAD

**Issue by mcarn** [Jaliborc/PetTracker#299]
**Running Software (issues missing this information will be deleted):**
 - Addon version: PetTracker-10.0-10.0.0.zip
 - Server patch: 10.0.0[46366] x64

 **Have you read the changelog? (please don't waste our time)**
Yes

**Describe the bug**
When challenging any pet tamer, the log below is displayed in a popup.
Once the tamer is defeated, the popup doesn't show when clicked again.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to any pet tamer( ex. [Major Payne](https://www.wowhead.com/npc=66675/major-payne) in Northrend)
2. Click on them
3. error popup

**Expected behaviour**
The popup shouldn't appear.

**Screenshots**
![image](https://user-images.githubusercontent.com/28352648/199431145-0d0011f5-5c23-49ed-8734-a765eb1aafdd.png)

**Error Logs**

> Message: ...ce/AddOns/PetTracker/addons/main/features/gossip.lua:28: attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)
> Time: Wed Nov  2 07:40:06 2022
> Count: 1
> Stack: ...ce/AddOns/PetTracker/addons/main/features/gossip.lua:28: attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)
> [string "@Interface/AddOns/PetTracker/addons/main/features/gossip.lua"]:28: in function <...ce/AddOns/PetTracker/addons/main/features/gossip.lua:23>
> [string "=[C]"]: in function `Update'
> [string "@Interface/FrameXML/GossipFrame.lua"]:66: in function `HandleShow'
> [string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
> [string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:39: in function <Interface/FrameXML/CustomGossipFrameBase.lua:37>
> 
> Locals: unit = "Creature-0-1465-571-51-66675-000061C9C3"
> id = 66675
> (*temporary) = nil
> (*temporary) = "attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)"
> Addon = <table> {
>  Gossip = <table> {
>  }
>  Enemy = <table> {
>  }
>  GetColor = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:66
>  Species = <table> {
>  }
>  Rival = <table> {
>  }
>  OnPetsChanged = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:53
>  Tracker = <table> {
>  }
>  Search = <function> defined @Interface/AddOns/PetTracker/addons/main/api/search.lua:26
>  RivalInfo = <table> {
>  }
>  Delay = <function> defined @Interface/AddOns/PetTracker/libs/MutexDelay-1.0/MutexDelay-1.0.lua:27
>  MultiTip = <table> {
>  }
>  RegisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
>  UnregisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:153
>  Rivals = <table> {
>  }
>  Base = <table> {
>  }
>  UnregisterSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:78
>  SpecieBreeds = <table> {
>  }
>  RegisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
>  sets = <table> {
>  }
>  OnBattle = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:57
>  AbilityButton = <table> {
>  }
>  OnEnable = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:26
>  Name = "PetTracker"
>  BattleSlot = <table> {
>  }
>  Objectives = Frame {
>  }
>  SendSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:82
>  SpeciePin = <table> {
>  }
>  NewModule = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:68
>  MaxLevel = 25
>  RivalPin = <table> {
>  }
>  Entity = <table> {
>  }
>  RivalOrder = <table> {
>  }
>  JournalSlot = <table> {
>  }
>  TextLine = <table> {
>  }
>  StablePin = <table> {
>  }
>  AbilityDisplay = <table> {
>  }
>  Breeds = <table> {
>  }
>  Pin = <table> {
>  }
>  Tag = "PETTRACKER_"
>  TextList = <table> {
>  }
>  MapCanvas = <table> {
>  }
>  ProgressBar = <table> {
>  }
>  SpecieStats = <table> {
>  }
>  MapSearch = <table> {
>  }
>  PetSlot = <table> {
>  }
>  TrackToggle = PetTrackerTrackToggle {
>  }
>  SendMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:59
>  RivalsJournal = PetTrackerRivalsJournal {
>  }
>  BattleRecord = <table> {
>  }
>  Stables = <table> {
>  }
>  UnregisterAllEvents = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
>  AbilityAction = <table> {
>  }
>  JournalMods = <table> {
>  }
>  RegisterSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:74
>  UnregisterAllMessages = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
>  Pet = <table> {
>  }
>  state = <table> {
>  }
>  Delaying = <function> defined @Interface/AddOns/PetTracker/libs/MutexDelay-1.0/MutexDelay-1.0.lua:46
>  Tooltips = <table> {
>  }
>  Predict = <table> {
>  }
>  MaxQuality = 6
>  Specie = <table> {
>  }
>  Ability = <table> {
>  }
>  UnregisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib

**Additional context**

**Comment by RudolphRiedel** [comment]
Came here to report the same, just got this with Zunta, south of Orgrimmar.

**Comment by Masumia** [comment]
Same here...

```
1x ...ce/AddOns/PetTracker/addons/main/features/gossip.lua:28: attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)
[string "@Interface/AddOns/PetTracker/addons/main/features/gossip.lua"]:28: in function <...ce/AddOns/PetTracker/addons/main/features/gossip.lua:23>
[string "=[C]"]: in function `Update'
[string "@Interface/FrameXML/GossipFrame.lua"]:66: in function `HandleShow'
[string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
[string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:39: in function <Interface/FrameXML/CustomGossipFrameBase.lua:37>

Locals:
unit = "Creature-0-4225-1116-14-87122-0000612FF5"
id = 87122
(*temporary) = nil
(*temporary) = "attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)"
Addon = <table> {
 SendSignal = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/WildAddon-1.0/WildAddon-1.0.lua:82
 Gossip = <table> {
 }
 SpeciePin = <table> {
 }
 UnregisterAllEvents = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
 Enemy = <table> {
 }
 sets = <table> {
 }
 state = <table> {
 }
 MaxLevel = 25
 GetColor = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:66
 Entity = <table> {
 }
 Species = <table> {
 }
 Delaying = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/MutexDelay-1.0/MutexDelay-1.0.lua:46
 JournalSlot = <table> {
 }
 Rival = <table> {
 }
 Tooltips = <table> {
 }
 OnPetsChanged = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:53
 NewModule = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/WildAddon-1.0/WildAddon-1.0.lua:68
 MultiTip = <table> {
 }
 MapCanvas = <table> {
 }
 Tracker = <table> {
 }
 Stables = <table> {
 }
 AbilityDisplay = <table> {
 }
 TextLine = <table> {
 }
 Search = <function> defined @Interface/AddOns/PetTracker/addons/main/api/search.lua:26
 StablePin = <table> {
 }
 RivalInfo = <table> {
 }
 Pin = <table> {
 }
 Delay = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/MutexDelay-1.0/MutexDelay-1.0.lua:27
 Tag = "PETTRACKER_"
 BattleSlot = <table> {
 }
 Pet = <table> {
 }
 AbilityAction = <table> {
 }
 TextList = <table> {
 }
 MaxQuality = 6
 UnregisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:153
 ProgressBar = <table> {
 }
 SpecieStats = <table> {
 }
 RivalPin = <table> {
 }
 MapSearch = <table> {
 }
 Specie = <table> {
 }
 PetSlot = <table> {
 }
 Ability = <table> {
 }
 Rivals = <table> {
 }
 Base = <table> {
 }
 Predict = <table> {
 }
 UnregisterSignal = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/WildAddon-1.0/WildAddon-1.0.lua:78
 Maps = <table> {
 }
 SpecieBreeds = <table> {
 }
 RegisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
 RegisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
 UnregisterAllMessages = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
 OnBattle = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:57
 Breeds = <table> {
 }
 AbilityButton = <table> {
 }
 OnEnable = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:26
 Name = "PetTracker"
 RivalOrder = <table> {
 }
 SendMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:59
 RegisterSignal = <function> defined @Interface/AddOns/Bagnon/common/Wildpants/libs/WildAddon-1.0/WildAddon-1.0.lua:74
 UnregisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:153
 Objectives = Frame {
 }
}
L = <table> {
 Switcher = "Switcher"
 Tutorial = <table> {
 }
 BreedExplanation = "Determines how stats gained at each level are distributed."
 FAQDescription = "These are the most frequently asked questions. To see the tutorials again, reset the addon settings using the "Defaults" button at the lower left corner."
 FilterSpecies = "Filter Species"
 Species = "Species"
 OptionsDescription = "These options allow you to toggle PetTracker general features on and off. Gotta catch them all!"
 Maximized = "Maximized"
 SpecieIconsTip = "If disabled, pets will be marked by their type instead of species when shown in the world and battle map."
 TotalRivals = "Total Rivals"
 Ninja = "Ninja"
 AddWaypoint = "Add Waypoint"
 AvailableBreeds = "Available Breeds"
 ShowStables = "Show Stables"
 Breed = "Breed"
 Source8 = "Promotion"
 TellMore = "Tell me more about yourself."
 AlertUpgrades = "Aler
```

**Comment by Gaviin1242** [comment]
Same, with the DMF trainers.


```
2x ...ce/AddOns/PetTracker/addons/main/features/gossip.lua:28: attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)
[string "@Interface/AddOns/PetTracker/addons/main/features/gossip.lua"]:28: in function <...ce/AddOns/PetTracker/addons/main/features/gossip.lua:23>
[string "=[C]"]: in function `Update'
[string "@Interface/FrameXML/GossipFrame.lua"]:66: in function `HandleShow'
[string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:47: in function `HandleOpenEvent'
[string "@Interface/FrameXML/CustomGossipFrameBase.lua"]:39: in function <Interface/FrameXML/CustomGossipFrameBase.lua:37>

Locals:
unit = "Creature-0-3778-974-11050-67370-0000676062"
id = 67370
(*temporary) = nil
(*temporary) = "attempt to call global 'GossipFrame_GetTitleButtonCount' (a nil value)"
Addon = <table> {
 SendSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:82
 Gossip = <table> {
 }
 SpeciePin = <table> {
 }
 UnregisterAllEvents = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
 Enemy = <table> {
 }
 sets = <table> {
 }
 state = <table> {
 }
 MaxLevel = 25
 GetColor = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:66
 Entity = <table> {
 }
 Species = <table> {
 }
 Delaying = <function> defined @Interface/AddOns/PetTracker/libs/MutexDelay-1.0/MutexDelay-1.0.lua:46
 JournalSlot = <table> {
 }
 Rival = <table> {
 }
 Tooltips = <table> {
 }
 OnPetsChanged = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:53
 NewModule = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:68
 MultiTip = <table> {
 }
 MapCanvas = <table> {
 }
 Tracker = <table> {
 }
 Stables = <table> {
 }
 AbilityDisplay = <table> {
 }
 TextLine = <table> {
 }
 Search = <function> defined @Interface/AddOns/PetTracker/addons/main/api/search.lua:26
 StablePin = <table> {
 }
 RivalInfo = <table> {
 }
 Pin = <table> {
 }
 Delay = <function> defined @Interface/AddOns/PetTracker/libs/MutexDelay-1.0/MutexDelay-1.0.lua:27
 Tag = "PETTRACKER_"
 BattleSlot = <table> {
 }
 Pet = <table> {
 }
 AbilityAction = <table> {
 }
 TextList = <table> {
 }
 MaxQuality = 6
 UnregisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:153
 ProgressBar = <table> {
 }
 SpecieStats = <table> {
 }
 RivalPin = <table> {
 }
 MapSearch = <table> {
 }
 Specie = <table> {
 }
 PetSlot = <table> {
 }
 Ability = <table> {
 }
 Rivals = <table> {
 }
 Base = <table> {
 }
 Predict = <table> {
 }
 UnregisterSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:78
 Maps = <table> {
 }
 SpecieBreeds = <table> {
 }
 RegisterMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
 RegisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:90
 UnregisterAllMessages = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:174
 OnBattle = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:57
 Breeds = <table> {
 }
 AbilityButton = <table> {
 }
 OnEnable = <function> defined @Interface/AddOns/PetTracker/addons/main/main.lua:26
 Name = "PetTracker"
 RivalOrder = <table> {
 }
 SendMessage = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:59
 RegisterSignal = <function> defined @Interface/AddOns/PetTracker/libs/WildAddon-1.0/WildAddon-1.0.lua:74
 UnregisterEvent = <function> defined @Interface/AddOns/AllTheThings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:153
 Objectives = Frame {
 }
}
L = <table> {
 Switcher = "Switcher"
 Tutorial = <table> {
 }
 BreedExplanation = "Determines how stats gained at each level are distributed."
 FAQDescription = "These are the most frequently asked questions. To see the tutorials again, reset the addon settings using the "Defaults" button at the lower left corner."
 FilterSpecies = "Filter Species"
 Species = "Species"
 OptionsDescription = "These options allow you to toggle PetTracker general features on and off. Gotta catch them all!"
 Maximized = "Maximized"
 SpecieIconsTip = "If disabled, pets will be marked by their type instead of species when shown in the world and battle map."
 TotalRivals = "Total Rivals"
 Ninja = "Ninja"
 AddWaypoint = "Add Waypoint"
 AvailableBreeds = "Available Breeds"
 ShowStables = "Show Stables"
 Breed = "Breed"
 Source8 = "Promotion"
 TellMore = "Tell me more about yourself."
 AlertUpgrades = "Alert for Upgrades"
 NoneCollected = "None Collected"
 EnemyTeam = "Enemy Team"
```

**Comment by fubaWoW** [comment]
same here

**Comment by GurliGebis** [comment]
The correct fix for this is in #312 - instead of checking if the function exists (which it never will), just change it to use the correct function instead.

**Comment by GurliGebis** [comment]
My bad, it seems like this is better until a better solution can be found.
